@kiva/kv-components/vue/KvSideSheet.vue

```diff
@@ -98,36 +118,86 @@ export default {
 			visible,
 			kvTrackFunction,
 			trackEventCategory,
+			expandEffect,
+			initialStyles,
 		} = toRefs(props);
 
 		const open = ref(false);
+		const animating = ref(false);
+
+		const modalStyles = ref({});
+
+		if (expandEffect.value) {
+			modalStyles.value = {
+				...initialStyles,
+				transition: 'none',
+			};
+		}
 
 		const closeSideSheet = () => {
 			open.value = false;
 			kvTrackFunction.value(trackEventCategory.value, 'click', 'side-sheet-closed');
+
+			if (expandEffect.value) {
+				animating.value = true;
+
+				modalStyles.value = {
+					...initialStyles.value,
+					transition: 'all 0.7s ease-in-out',
+				};
+			}
+
 			setTimeout(() => {
 				emit('side-sheet-closed');
+				animating.value = false;
 			}, '700');
 		};
 
 		const goToLink = () => {
 			emit('go-to-link');
 		};
 
+		const onTransitionEnd = () => {
+			animating.value = false;
+		};
+
 		watch(visible, () => {
 			if (visible.value) {
 				setTimeout(() => {
 					open.value = true;
 				}, '300');
+
+				if (expandEffect.value) {
+					animating.value = true;
```

Comment:
Right before the side sheet opens, the original button disappears, the animation waits a second, then it starts. We should have this be smoother, if possible. Like can the animation start without that delay so it doesn't look like the button disappears first?